src/MLOps.NET/MLLifeCycleManager.cs

```diff
@@ -8,15 +8,21 @@ namespace MLOps.NET
     ///<inheritdoc cref="IMLLifeCycleManager"/>
     public class MLLifeCycleManager : IMLLifeCycleManager
     {
+        internal MLLifeCycleManager(IMetaDataStore metaDataStore, IModelRepository modelRepository)
+        {
+            MetaDataStore = metaDataStore ?? throw new ArgumentNullException(nameof(metaDataStore));
+            ModelRepository = modelRepository ?? throw new ArgumentNullException(nameof(modelRepository));
+        }
+
         /// <summary>
         /// Repository for model metadata 
         /// </summary>
-        public IMetaDataStore MetaDataStore { get; set; }
+        public IMetaDataStore MetaDataStore { get; }
```

Comment:
This is great, lets take it one step further and make them private member variables.

Reply:
done - I used `lowerCamelCase` for private field names - let me know if you prefer something different (underscore prefix, etc)